### ISSUE TITLE
Error after building on ESXi due to missing output directory

### ISSUE THREAD

**Issue by voiprodrigo** [mitchellh/packer#4028]
Hello,

I'm using Packer 0.10.2 on CentOS 7.2x64 to build VM's on ESXi 6.0.2, using the vmware-iso builder. In the final stages I get this error:

```
2016/10/19 16:54:12 packer: 2016/10/19 16:54:12 starting remote command: vim-cmd vmsvc/reload 46
2016/10/19 16:54:12 packer: 2016/10/19 16:54:12 remote command exited with '0': vim-cmd vmsvc/reload 46
2016/10/19 16:54:12 packer: 2016/10/19 16:54:12 opening new ssh session
2016/10/19 16:54:12 packer: 2016/10/19 16:54:12 starting remote command: vim-cmd vmsvc/power.getstate 46
2016/10/19 16:54:13 packer: 2016/10/19 16:54:13 remote command exited with '0': vim-cmd vmsvc/power.getstate 46
==> centos72: Keeping virtual machine registered with ESX host (keep_registered = true)
2016/10/19 16:54:13 ui: ==> centos72: Keeping virtual machine registered with ESX host (keep_registered = true)
2016/10/19 16:54:13 packer: 2016/10/19 16:54:13 Deleting floppy disk: /tmp/packer872343441
2016/10/19 16:54:13 ui error: Build 'centos72' errored: lstat output-centos72: no such file or directory
2016/10/19 16:54:13 Builds completed. Waiting on interrupt barrier...
2016/10/19 16:54:13 machine readable: error-count []string{"1"}
2016/10/19 16:54:13 ui error:
==> Some builds didn't complete successfully and had errors:
2016/10/19 16:54:13 machine readable: centos72,error []string{"lstat output-centos72: no such file or directory"}
2016/10/19 16:54:13 ui error: --> centos72: lstat output-centos72: no such file or directory
2016/10/19 16:54:13 ui:
==> Builds finished but no artifacts were created.
2016/10/19 16:54:13 waiting for all plugin processes to complete...
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
Build 'centos72' errored: lstat output-centos72: no such file or directory

==> Some builds didn't complete successfully and had errors:
--> centos72: lstat output-centos72: no such file or directory

==> Builds finished but no artifacts were created.
2016/10/19 16:54:13 /usr/local/bin/packer: plugin process exited
```

I can prevent the error if before launching packet, I create output-centos72 on the current working directory. In which case the output is:

```
==> centos72: Keeping virtual machine registered with ESX host (keep_registered = true)
2016/10/19 17:11:20 ui: ==> centos72: Keeping virtual machine registered with ESX host (keep_registered = true)
2016/10/19 17:11:20 packer: 2016/10/19 17:11:20 Deleting floppy disk: /tmp/packer926348574
2016/10/19 17:11:20 ui: Build 'centos72' finished.
2016/10/19 17:11:20 Builds completed. Waiting on interrupt barrier...
2016/10/19 17:11:20 ui:
==> Builds finished. The artifacts of successful builds are:
2016/10/19 17:11:20 machine readable: centos72,artifact-count []string{"1"}
Build 'centos72' finished.

==> Builds finished. The artifacts of successful builds are:
2016/10/19 17:11:20 machine readable: centos72,artifact []string{"0", "builder-id", "mitchellh.vmware-esx"}
2016/10/19 17:11:20 machine readable: centos72,artifact []string{"0", "id", "VM"}
2016/10/19 17:11:20 machine readable: centos72,artifact []string{"0", "string", "VM files in directory: output-centos72"}
2016/10/19 17:11:20 machine readable: centos72,artifact []string{"0", "files-count", "0"}
2016/10/19 17:11:20 machine readable: centos72,artifact []string{"0", "end"}
2016/10/19 17:11:20 ui: --> centos72: VM files in directory: output-centos72
2016/10/19 17:11:20 waiting for all plugin processes to complete...
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
--> centos72: VM files in directory: output-centos72
2016/10/19 17:11:20 /usr/local/bin/packer: plugin process exited
```

Obviously, there's nothing inside output-centos72, as the VM is deployed in ESXi.

Sorry if this is already known.

**Comment by rickard-von-essen** [comment]
Please supply the information requested in the issue template:
- Packer version (`packer version`)
- Host platform (`uname -a` etc.)
- Debug log output from `PACKER_LOG=1 packer build template.json`.
  Please paste this in a [gist](https://gist.github.com).
- The _simplest example template and scripts_ needed to reproduce the bug.
  Include these in your [gist](https://gist.github.com).

This is most likely fixed in #3050 which will be out in 0.11.0.

**Comment by voiprodrigo** [comment]
Thanks. I've read the description of #3050 and it does seem related. I could add the extra required info but doesn't seem necessary.
